### PR TITLE
feature: support Oracle Batch Insert

### LINKED ISSUE #7607: Oracle Batch Insert Not Support
### Check Ahead

- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

- [ ] I am willing to try to fix this bug myself.

### Ⅰ. Issue Description

i have a sql like: **insert all  into a (id) values(1)  into a (id) values(2) SELECT 1 FROM DUAL**

it will parse to OracleMultiInsertStatement

but io.seata.sqlparser.druid.DruidSQLRecognizerFactoryImpl not support OracleMultiInsertStatement type

Image

### Ⅱ. Describe what happened

_No response_

### Ⅲ. Describe what you expected to happen

_No response_

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

_No response_

### Ⅴ. Anything else we need to know?

_No response_

### Ⅵ. Environment

seata version:2.0
druid version:1.2.8

### PR DESCRIPTION
**Pull request by maple525866**
- [x] I have read the [CONTRIBUTING.md](https://github.com/apache/incubator-seata/blob/2.x/CONTRIBUTING.md) guidelines.
- [x] I have registered the PR [changes](https://github.com/apache/incubator-seata/tree/2.x/changes).

### Ⅰ. Describe what this PR did
1. Problem Background
Seata cannot support Oracle's INSERT ALL batch insert statements. For example, `insert all into a (id) values(1) into a (id) values(2) SELECT 1 FROM DUAL` will be parsed as `OracleMultiInsertStatement`, but `DruidSQLRecognizerFactoryImpl` does not support this statement type.

2. Solution
- Extend `DruidSQLRecognizerFactoryImpl` to add support for `OracleMultiInsertStatement`
- Extend `OracleOperateRecognizerHolder` to add methods specifically for handling multiple inserts
- Create `OracleMultiInsertRecognizer` and use `ast.getEntries()` to obtain insert items. It supports parsing multiple INSERT INTO clauses and correctly handles various data types (NULL, parameter placeholders, sequences, etc.)

3. Verification Functionality  -> Created OracleMultiInsertRecognizerTest containing 11 test cases:
- Basic functional testing: SQL type, table name, table alias
- Column processing testing: Insert column retrieval, empty column detection
- Data row testing: Basic data, NULL values, parameter placeholders, primary key processing
- Complex scenario testing: Multi-table inserts, complex statements, edge cases
- Interface consistency testing: Ensures compliance with the Seata interface specification

### Ⅱ. Does this pull request fix one issue?

fixes #7607 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
UT have been added

### Ⅳ. Describe how to verify it
mvn clean test

### Ⅴ. Special notes for reviews
Currently, only single-table Oracle INSERT batching is supported. I don't believe multi-table support is necessary because branching transactions can be useful.❤️

### CODE REVIEW

File: sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/DruidSQLRecognizerFactoryImpl.java [hunk 1]
```diff
@@ -72,6 +74,10 @@ public List<SQLRecognizer> create(String sql, String dbType) {
                 recognizer = recognizerHolder.getDeleteRecognizer(sql, sqlStatement);
             } else if (sqlStatement instanceof SQLSelectStatement) {
                 recognizer = recognizerHolder.getSelectForUpdateRecognizer(sql, sqlStatement);
+            } else if (sqlStatement instanceof OracleMultiInsertStatement) {
+                // Use specialized methods to handle Oracle bulk inserts
+                recognizer =
+                        ((OracleOperateRecognizerHolder) recognizerHolder).getMultiInsertRecognizer(sql, sqlStatement);
```
Comment (YoWuwuuuw) [inline sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/DruidSQLRecognizerFactoryImpl.java:84]: OracleMultiInsertStatement has two syntaxes: `Insert All` and `Insert First`. If your implementation class does not support `Insert First` syntax, you should throw NotSupportYetException in `isSqlSyntaxSupports()` of BaseOracleRecognizer.

File: sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    @Override
+    public SQLType getSQLType() {
+        return SQLType.INSERT;
+    }
+
+    @Override
+    public String getTableAlias() {
+        // Oracle Multi Insert usually only one table is involved, take the table alias of the first inserted item
+        if (!CollectionUtils.isEmpty(ast.getEntries())) {
+            OracleMultiInsertStatement.Entry firstEntry = ast.getEntries().get(0);
```
Comment (YoWuwuuuw) [inline sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java]: There may be different table names or aliases. If you directly use the first entry to get it, I think this will cause some problems.

File: sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean insertColumnsIsEmpty() {
+        if (!CollectionUtils.isEmpty(ast.getEntries())) {
+            OracleMultiInsertStatement.Entry firstEntry = ast.getEntries().get(0);
```
Comment (YoWuwuuuw) [inline sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java]: Same as above.

File: sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        return null;
+    }
+
+    @Override
+    public List<List<Object>> getInsertRows(Collection<Integer> primaryKeyIndex) {
+        List<List<Object>> allRows = new ArrayList<>();
+
+        if (!CollectionUtils.isEmpty(ast.getEntries())) {
+            for (OracleMultiInsertStatement.Entry entry : ast.getEntries()) {
+                if (entry instanceof OracleMultiInsertStatement.InsertIntoClause) {
```
Comment (YoWuwuuuw) [inline sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java:199]: `OracleMultiInsertStatement.Entry` is divided into `ConditionalInsertClause` and `InsertIntoClause`. If it is the former, will some data be missed?

File: sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java
Comment (YoWuwuuuw) [inline sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java:1]: By default, only the `insert all` syntax of Oracle Batch Insert and the same column situation is supported. The same effect can be achieved by adding multiple values ​​clauses to a normal insert. Are there plans to support more in the future?

Comment (maple525866) [reply sqlparser/seata-sqlparser-druid/src/main/java/org/apache/seata/sqlparser/druid/oracle/OracleMultiInsertRecognizer.java:1]: Currently, we only support batch insert of multiple columns (same columns) in a single table. We will continue to support this in the future.

**Comment by maple525866** [comment]
First, thanks to @YoWuwuuuw for the bug report.❤️

For now, I think we can start with basic support, like this pull request, which supports batch inserts for multiple columns in a single table.

However, for other scenarios, such as support for insert-first syntax and ConditionalInsertClause, further support can be implemented through other pull requests.

For multi-table scenarios, I don't think support is necessary at this time.

**Comment by maple525866** [comment]
@funky-eyes PTAL❤️  If you have any further questions please let me know

**Review by funky-eyes — APPROVED** [review]
LGTM

**Comment by YvCeung** [comment]
Well done. The code format verification failed due to other PR(#7643 ) and will be fixed later.

**Comment by YvCeung** [comment]
LGTM
